src/main/java/org/elasticsearch/index/mapper/geo/GeoShapeFieldMapper.java

```diff
@@ -273,40 +277,38 @@ public void merge(Mapper mergeWith, MergeContext mergeContext) throws MergeMappi
             return;
         }
         final GeoShapeFieldMapper fieldMergeWith = (GeoShapeFieldMapper) mergeWith;
-        if (!mergeContext.mergeFlags().simulate()) {
-            final PrefixTreeStrategy mergeWithStrategy = fieldMergeWith.defaultStrategy;
+        final PrefixTreeStrategy mergeWithStrategy = fieldMergeWith.defaultStrategy;
```

Comment:
Did you mean to make this "if (simulate) return" instead of removing the simulate check altogether?